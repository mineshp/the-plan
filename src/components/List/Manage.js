import React from 'react';
import { Container } from 'semantic-ui-react';
import ManageList from '../../HOC/List/ManageList';

const Manage = () => (
    <Container>
        <ManageList />
    </Container>
);

export default Manage;
