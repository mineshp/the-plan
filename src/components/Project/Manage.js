import React from 'react';
import ManageProject from '../../HOC/Project/ManageProject.js';
import LeftHandMenu from '../Shared/LeftHandMenu.js';
import { Container, Grid } from 'semantic-ui-react';

const Manage = () => (
  <Container>
    <Grid>
      <Grid.Column width={4}>
        <LeftHandMenu />
      </Grid.Column>
      <Grid.Column width={12}>
        <ManageProject />
      </Grid.Column>
    </Grid>
  </Container>
)

export default Manage;