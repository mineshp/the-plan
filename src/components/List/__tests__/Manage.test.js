import React from 'react';
import renderer from 'react-test-renderer';
import Manage from '../Manage';

const mockListAll = [
    {
        _id: '5992092f66a7043f2598c88e',
        projects: [
            {
                id: 'abc123',
                name: 'biology'
            },
            {
                id: 'xyz123',
                name: 'chemistry'
            }
        ],
        listName: 'HumanBody',
        createdDate: '2016-05-18T16:00:00Z',
        updatedDate: '2016-05-18T16:00:00Z',
        headings: [
            {
                position: 1,
                id: '1',
                name: 'Name'
            },
            {
                position: 2,
                id: '2',
                name: 'Created'
            }
        ],
        items: [
            {
                updatedDate: '2016-05-18T16:00:00Z',
                createdDate: '2016-05-18T16:00:00Z',
                position: 1,
                id: '1',
                name: 'Head'
            },
            {
                updatedDate: '2016-05-18T16:00:00Z',
                createdDate: '2016-05-18T16:00:00Z',
                position: 2,
                id: '2',
                name: 'Shoulder'
            }
        ]
    }
];

const mockResponse = (status, statusText, response) => new window.Response(response, {
    status,
    statusText,
    headers: {
        'Content-type': 'application/json'
    }
});

describe('Manage Lists', () => {
    beforeEach(() => {
        window.fetch = jest.fn().mockImplementation(() =>
            Promise.resolve(mockResponse(200, null, JSON.stringify(mockListAll))));
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Manage />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
