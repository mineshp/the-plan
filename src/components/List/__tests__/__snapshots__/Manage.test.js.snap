// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage Lists renders correctly 1`] = `
<div
  className="ui container"
>
  <div
    className="List main"
  >
    <table
      className="ui celled striped table"
    >
      <thead
        className=""
      >
        <tr
          className=""
        >
          <th
            className=""
          >
            Name
          </th>
          <th
            className=""
          >
            Created
          </th>
          <th
            className=""
          >
            Last Updated
          </th>
          <th
            className=""
          >
            Projects
          </th>
          <th
            className=""
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        className=""
      />
    </table>
  </div>
</div>
`;
